src/topotoolbox/stream_object.py

```diff
@@ -1186,26 +1186,30 @@ def crslin(self, dem, k = 1, mingradient=0.0, attachheads=False, attachtomin=Fal
         values = np.array(
             [2/((xi-xj)*(xk-xj)), -2/((xk-xi)*(xi-xj)), 2/((xk-xi)*(xk-xj))])
 
-        # Sparse kvxopt second derivative matrix (C)
-        c = spmatrix(values.T.flatten().tolist(), rowix.flatten(
-        ).tolist(), colix.flatten().tolist(), (nrrows, nr))
+        # Second derivative matrix (C)
+        c = sp.csc_array((values.flatten(), (rowix.flatten(), colix.flatten())), (nrrows, nr))
```

Comment:
You dropped the needed transpose when you converted this to `csc_array`. You need the change:

```diff
@@ -1187,7 +1187,7 @@ class StreamObject():
             [2/((xi-xj)*(xk-xj)), -2/((xk-xi)*(xi-xj)), 2/((xk-xi)*(xk-xj))])
 
         # Second derivative matrix (C)
-        c = sp.csc_array((values.flatten(), (rowix.flatten(), colix.flatten())), (nrrows, nr))
+        c = sp.csc_array((values.T.flatten(), (rowix.flatten(), colix.flatten())), (nrrows, nr))
 
         #########################################
         # Compute s parameter (equation A7)
```